### PR TITLE
PR75 updated with Check Support 

### PR DESCRIPTION
**Pull request by mccv1r0**

### CODE REVIEW

File: plugins/meta/firewall/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# firewall plugin
+
+## Overview
+
+This plugin creates firewall rules to allow traffic to/from container IP address via the host network .
+It does not create any network interfaces and therefore does not set up connectivity by itself.
+It is only useful when used in addition to other plugins.
```
Comment (squeed) [inline plugins/meta/firewall/README.md]: "It is intended to be used as a chained plugin."

File: plugins/meta/firewall/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+This plugin supports multiple firewall backends that implement the desired functionality.
+Available backends include `iptables` and `firewalld` and may be selected with the `backend` key.
+If no `backend` key is given, the plugin will use firewalld if the service exists on the D-Bus system bus.
+If no firewalld service is found, it will fall back to iptables.
+
+When the `iptables` backend is used, the above example will create two new iptables chains in the `filter` table and add rules that allow the given interface to send/receive traffic.
+When the `firewalld` backend is used, the above example will place the `cni0` interface into firewalld's `trusted` zone, allowing it to send/receive traffic.
+
+
```
Comment (squeed) [inline plugins/meta/firewall/README.md]: Like the portmap plugin, could you add an example and detail the iptables rules that would be created?

File: plugins/meta/firewall/firewall.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	// IptablesAdminChainName is an optional name to use instead of the default
+	// admin rules override chain name that includes the interface name.
+	IptablesAdminChainName string `json:"iptablesAdminChainName,omitempty"`
+
+	// FirewalldZone is an optional firewalld zone to place the interface into.  If
+	// the firewalld backend is used but the zone is not given, it defaults
+	// to 'trusted'
+	FirewalldZone string `json:"firewalldZone,omitempty"`
+
+	//RawPrevResult map[string]interface{} `json:"prevResult,omitempty"`
```
Comment (squeed) [inline plugins/meta/firewall/firewall.go]: nit: can you remove these commented-out members?

File: plugins/meta/firewall/firewall.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	// Runtime errors are ignored
+	if err := backend.Del(conf, result); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, "TODO")
```
Comment (squeed) [inline plugins/meta/firewall/firewall.go]: I'm not sure that we support all versions. Does the plugin need the prev result on DEL to work? If so, then we only support 0.4.0.

Comment (mccv1r0) [reply plugins/meta/firewall/firewall.go]: There is a runtime check which verified that prev result was supplied (on both del and check.)  But I updated this call to use ```version.PluginSupports("0.4.0")```

**Comment by mrostecki** [comment]
Nice, we finally made it. :+1:

**Pull request merged**